### PR TITLE
fix(convict): add optional second parameter

### PR DESCRIPTION
**Pull request by petermetz**
Needed for injecting custom process.env or process.args parameters to be used by convict which is sorely needed when testing code that uses convict for it's configuration parsing.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mozilla/node-convict/blob/9d9070966f3dc9dc269895b09d5b4a4777a44ccb/packages/convict/src/main.js#L454
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/convict/index.d.ts [hunk 1]
```diff
@@ -199,7 +200,7 @@ interface convict {
     addFormat(format: convict.Format): void;
     addFormats(formats: { [name: string]: convict.Format }): void;
     addParser(parsers: convict.Parser | convict.Parser[]): void;
-    <T>(config: convict.Schema<T> | string): convict.Config<T>;
+    <T>(config: convict.Schema<T> | string, options?: { env?: any, args?: string[] }): convict.Config<T>;
```
Comment (vesse) [inline types/convict/index.d.ts:203]: I would prefer defining separate type and use the same typedef for `env` that `process.env` has, ie.

```
/// <reference types="node" />

...

    type Options = {
        env?: NodeJS.ProcessEnv;
        args?: string[];
    }

...

    <T>(config: convict.Schema<T> | string, opts?: convict.Options): convict.Config<T>;
```

**Review by vesse — CHANGES_REQUESTED** [review]

File: types/convict/index.d.ts (hunk 1 above)
Comment (petermetz) [reply types/convict/index.d.ts]: @vesse Thanks for the quick review! I agree, but switched out the type to an interface, I hope that's OK. I prefer to use interfaces when not specifically forced to use a type. Can't remember for sure but I may have read it in the official Typescript docs that this is a generally recommended approach.

**Review by vesse — APPROVED** [review]

**Comment by xamgore** [comment]
Ready to merge

**Comment by petermetz** [comment]
Ready to merge

**Pull request merged**